CKSIncrementalStore/CKSIncrementalStoreSyncOperation.swift

```diff
@@ -639,7 +639,7 @@ class CKSIncrementalStoreSyncOperation: NSOperation {
         if self.localStoreMOC!.hasChanges
         {
             var error:NSErrorPointer = nil
-            self.localStoreMOC!.save(error)
+            self.localStoreMOC!.save(error) // TODO: saving one object at a time (or recovering and saving as much as possible) would be more robust (and catching exceptions during all the above). Otherwise a failure will result in the entire change set being lost. Also, it probably should revert/rollback on failure also.
```

Comment:
The case you just talked about is left to the user of this library to handle. This should be taken in notice by the developer using this library to consider these cases and set validation rules that do not break any previous versions of the app. That is why Its said that one must avoid these types of hard validation rules because they do not scale well :)